File: partition_pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        l1 = []
        l2 = []
        c = 0
        for i in nums:
            if i < pivot:
                l1.append(i)
            elif i == pivot:
                c += 1
            else:
                l2.append(i)
        return l1 + ([pivot] * c) + l2

File: Go_version/leetcode/partitionPivot.go
package leetcode

func PivotArray(nums []int, pivot int) []int {
	a := []int{}
	l1 := []int{}
	l2 := []int{}
	c := 0
	for i := range len(nums) {
		if nums[i] < pivot {
			l1 = append(l1, nums[i])
		} else if nums[i] > pivot {
			l2 = append(l2, nums[i])
		} else {
			c++
		}
	}
	a = append(a, l1...)
	pivotSlice := make([]int, c)
	for i := range pivotSlice {
		pivotSlice[i] = pivot
	}
	a = append(a, pivotSlice...)
	a = append(a, l2...)
	return a
}
